### DISCUSSION TITLE
Blog 5: Hello World

### DISCUSSION THREAD

**Discussion by Max-Rohrhirsch** [DHBW-UMOC/UMOC#9] [Announcements]
# Endlich Programmieren
Die Aufgabe war es, einen Prototypen zu programmieren. Hierfür haben wir mit dem backend angefangen. 
Für die Datenbank nutzen wir `SQLight` da es einfach aufzusetzen ist und File based ist. 
Um normale Schnittstellen zu erstellen, haben wir Flask verwendet.
Da ein Chat dauerhaft aktualisiert werden soll, und man nicht immer die ganze Seite neu laden möchte und zu schauen, ob sich etwas geändert hat, brauchen wir Websockets. Diese haben wir mit flask_socketio umgesetzt.
Wir haben auch bereits einen Test geschrieben, der die Request überprüft, da wir noch kein Frontend haben.

## Features
Da das nur ein Prototyp bisher ist und dem Projekt Struktur verleihen soll, kann sich bisher nur einloggen und bekommt einen Gruß vom Programm.

## Wo ist der Code?
Im Moment sind in unser Organisation 3 Repositorys. 
1. UMOC: Hier ist nur die Discussion drin. Zwar ist auch noch code vorhanden, dieser ist aber nicht mehr Relevant.
2. UMOC-Backend: Hier ist das Backend mit unserem Prototyp.
3. UMOC-Frontend: Hier wird in Zukunft das Frontend hinzugefügt.

**Comment by TimHitschfeld** [comment]
Es ist schön zu sehen, dass das Projekt in die Praktische Phase gestartet ist und Ihr euch mit den Entwicklungstools vertraut gemacht habt. Wir freuen uns auf den ersten Prototypen, mit welchem man die ersten Grundfunktionen ausprobieren kann. Ein paar Bilder zur Veranschaulichung hätten euren Blogpost gut abgerundet.

Viele Grüße,
Euer TripAhead Team /TH
